src/debugger/handler_dbgp.c

```diff
@@ -2235,26 +2352,7 @@ static int xdebug_dbgp_parse_option(xdebug_con *context, char* line, int flags,
 		command = lookup_cmd(cmd);
 
 		if (command) {
-			if (command->cont) {
-				XG_DBG(status) = DBGP_STATUS_RUNNING;
-				XG_DBG(reason) = DBGP_REASON_OK;
-			}
-			XG_DBG(lastcmd) = command->name;
-			if (XG_DBG(lasttransid)) {
-				xdfree(XG_DBG(lasttransid));
-			}
-			XG_DBG(lasttransid) = xdstrdup(CMD_OPTION_CHAR('i'));
-			if (XG_DBG(status) != DBGP_STATUS_STOPPING || (XG_DBG(status) == DBGP_STATUS_STOPPING && command->flags & XDEBUG_DBGP_POST_MORTEM)) {
-				command->handler((xdebug_xml_node**) &retval, context, args);
-				ret = command->cont;
-			} else {
-				error = xdebug_xml_node_init("error");
-				xdebug_xml_add_attribute_ex(error, "code", xdebug_sprintf("%lu", XDEBUG_ERROR_COMMAND_UNAVAILABLE), 0, 1);
-				ADD_REASON_MESSAGE(XDEBUG_ERROR_COMMAND_UNAVAILABLE);
```

Comment:
The whole check for availability seems gone now. Some of this (such as the lastcmd stuft) is now done in every handler, but why do it there when it can be done once here? I am probably misunderstanding so I am hoping you can explain this to me when we chat about it.